extensions/mentions/js/src/forum/utils/textFormatter.js

```diff
@@ -49,3 +49,21 @@ export function filterGroupMentions(tag) {
 
   tag.invalidate();
 }
+
+export function filterTagMentions(tag) {
+  if (app.initializers.has('flarum-tags')) {
+    const tagModel = app.store.getById('tags', tag.getAttribute('id'));
+
+    if (tagModel) {
+      tag.setAttribute('tagname', extractText(tagModel.name()));
+      tag.setAttribute('icon', tagModel.icon());
+      tag.setAttribute('color', tagModel.color());
+      tag.setAttribute('slug', tagModel.slug());
+      tag.setAttribute('class', isDark(tagModel.color()) ? 'TagMention--light' : 'TagMention--dark');
```

Comment:
There is a little confusion here, it should be the contrary :)
Cf. https://github.com/flarum/framework/pull/3653

```suggestion
      tag.setAttribute('class', isDark(tagModel.color()) ? 'TagMention--dark' : 'TagMention--light');
```

Reply:
In actual usage, I've found the opposite. See my [dev forum](https://dev.morland.it/d/415-tag-mention-tests) for a live version of this..

Reply:
I know it's working perfectly well, this is not a dev problem I suppose. :v: What I meant is just that the formal logic of `isDark` usage is to ask `is this a dark color ?`, and if it's `true`, to add a dark tag, on which we apply the `text-on-dark` rule. This is just a matter of consistency about the usage of this `isDark` utility across the project

But maybe there is something I don't understand in your case, I'll try to test your branch!

Reply:
Maybe it would be simpler to think about all this with https://github.com/flarum/framework/pull/3653 merged. We could then try to find a homogeneous and common way of operating with those contrast issues :)